lib/viewing/RenderableView.js

```diff
@@ -58,10 +54,38 @@ var RenderableView = _.extend({},
             }.bind(this));
         },
 
+        generateViewPlaceholders: function() {
+            var unplacedCount = this.childViewCount() - this.unrenderedChildViewCount();
+
+            // [wawjr3d] Do this before return so data.views is always an object
+            this.viewPlaceholders = new Array(unplacedCount);
+
+            if (!unplacedCount) {
+                return;
+            }
+
+            this.foreachChildView(function(viewRelationship, identifier) {
+                if (viewRelationship.hasBeenPlaced()) {
+                    return;
+                }
+
+                var placeholderId = VIEW_PLACEHOLDER_PREFIX + unplacedCount;
+
+                this.viewPlaceholders[identifier] = viewPlaceholder(placeholderId);
+                viewRelationship.andReplace("#" + placeholderId);
+
+                unplacedCount = unplacedCount - 1;
```

Comment:
this will cause the compiler to hold on to the parent scope, can we do it differently?